twenty-first/src/math/bfield_codec.rs

```diff
@@ -48,6 +50,9 @@ pub enum BFieldCodecError {
     #[error("invalid length indicator")]
     InvalidLengthIndicator,
 
+    #[error("trailing zeros in encoded polynomial")]
+    TrailingZerosInPolynomialCoefficients,
+
```

Comment:
In my opinion, this error is so specific to polynomials, it does not belong in the general `BFieldCodecError`. I feel it is cleaner to create a new `enum` that transparently uses `BFieldCodecError` as one of its variants.